### PR TITLE
Add Orion, modulemaps

### PR DESCRIPTION
**Pull request by kabiroberai**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This PR adds Orion.framework + modulemaps for Swift support

Does this close any currently open issues?
------------------------------------------
Nope

Any relevant logs, error output, etc?
-------------------------------------
Nope

Any other comments?
-------------------
Nope

Where has this been tested?
---------------------------
**Operating System:** macOS 12.4, Ubuntu 20.04

**Platform:** macOS, Linux

**Target Platform:** iOS

**Toolchain Version:** Xcode 13.4 on macOS, my latest toolchain on Linux

**SDK Version:** iOS 15.5

### CODE REVIEW

File: libundirect.tbd [hunk 1]
```diff
@@ -0,0 +1,19 @@
+--- !tapi-tbd-v3
+archs:           [ armv7, armv7s, arm64, arm64e ]
+uuids:           [ 'armv7: D740B3BA-C3E0-3358-86BB-8B1A8CB6A0F1', 'armv7s: D2A75695-3A8E-3C7D-A2A3-BCC5872C9791', 
+                   'arm64: 14D66645-F0E5-35AA-9512-E162A35A3929', 'arm64e: A1CB8039-5301-3312-8EB2-6DEC62A54106' ]
+platform:        ios
+flags:           [ not_app_extension_safe ]
+install-name:    '/usr/lib/libundirect.dylib'
```
Comment (elihwyma) [inline libundirect.tbd]: If https://github.com/theos/theos/pull/687 is merged this should be changed to "@rpath/libundirect.dylib"

**Review by elihwyma — CHANGES_REQUESTED** [review]

**Review by vip-caesar — APPROVED** [review]

File: Orion.framework/Headers/Orion.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#import <Foundation/Foundation.h>
+
+FOUNDATION_EXPORT double OrionVersionNumber;
+FOUNDATION_EXPORT const unsigned char OrionVersionString[];
```
Comment (leptos-null) [inline Orion.framework/Headers/Orion.h:4]: nit: May we remove these? They are not defined by the module

File: libundirect.tbd (hunk 1 above)
Comment (leptos-null) [reply libundirect.tbd]: Haven't looked at the details, but I see that the linked PR is closed. Are we able to resolve this thread?

**Review by leptos-null — APPROVED** [review]
Looks good to me. I believe the open thread is not a blocker. Additionally my "nit" comment is a common issue caused by Xcode generated code that rarely leads to errors in practice.

File: Orion.framework/Headers/Orion.h (hunk 2 above)
Comment (kabiroberai) [reply Orion.framework/Headers/Orion.h:4]: will remove in the next Orion update, I want to keep this in sync with CI

Comment (kirb) [reply Orion.framework/Headers/Orion.h:4]: @leptos-null These are generated by the compiler (or maybe Xcode) using the module name as a prefix I believe

File: libundirect.tbd (hunk 1 above)
Comment (kabiroberai) [reply libundirect.tbd]: Removed libundirect since it's orthogonal to this PR

**Pull request merged**